### ISSUE TITLE
Payment Methods inconsistencies when using PaymentSheet in deferred mode

### ISSUE THREAD

**Issue by pgiacomo69** [flutter-stripe/flutter_stripe#1837]
When using PaymentSheet in 'Deferred Mode' the allowed payment methods can't come from intent, because the intent will be generated only after the user has choosen a payment method. Payment methods that will be shown should are those selected in Stripe Settings. From code we should be able to further filtering allowed payment methods by giving an array when istantiating `IntentConfiguration`.

With current code we have two problems:

1. In iOS, payment methods like iDeal and PayPayl won't be shown
2. In Android all configured payment methods are shown, regardless of wich values we put in `paymentMethodTypes` array in `intentConfiguration`

**To Reproduce**
1. Enable  PayPal, an other payment types in your Stripe account
2. Add `paymentMethodTypes: ['card','paypal'],` in `IntentConfiguration` at 'initPaymentSheet' method of payment_sheet_deffered_screen.dart file
3. Run the example App
4. Launch 'Single Step (deferred payment)' demo from 'Payment Sheet' group
7. In iOS: Observe that only 'Card' Payment is shown
8. In Android: Observe that all configured payment methods are shown, even if not present in `paymentMethodTypes` array.

Solution:

1. iOS: A return Url should be configured when configuring the payment sheet
2. Android: JSONArray is missing from possible tipes in ReadableMap.java from stripe_android package

A PR will be submitted to address this issue.
